d2bs/kolbot/libs/common/Town.js

```diff
@@ -67,13 +67,18 @@ var Town = {
 
 	// Do town chores
 	doChores: function (repair = false) {
+		var i,
+			cancelFlags = [0x01, 0x02, 0x04, 0x08, 0x14, 0x16, 0x0c, 0x0f, 0x19, 0x1a],
+			npc = getUnit(1, "warriv");
+
 		if (!me.inTown) {
 			this.goToTown();
+		} else if (me.area === 40 && npc && getDistance(me, npc) < 50) {
+			if (npc.openMenu()) {
+				Misc.useMenu(0x0D37);
+			}
```

Comment:
While using warriv to travel to A1 for chores may save time in common cases, it changes behavior for bots that may expect to be on current act for other tasks. Most definitely this isn't desired behavior for bots on normal difficulty because pots are lower tiered in a1.